src/librustc_codegen_llvm/common.rs

```diff
@@ -259,10 +259,23 @@ impl ConstMethods<'tcx> for CodegenCx<'ll, 'tcx> {
                 let base_addr = match alloc_kind {
                     Some(GlobalAlloc::Memory(alloc)) => {
                         let init = const_alloc_to_llvm(self, alloc);
+                        let formatted_id = if self.sess().fewer_names() {
+                            None
+                        } else {
+                            Some(format!("{:?}", ptr.alloc_id))
+                        };
                         if alloc.mutability == Mutability::Mut {
-                            self.static_addr_of_mut(init, alloc.align, None)
+                            self.static_addr_of_mut(
+                                init,
+                                alloc.align,
+                                formatted_id.as_ref().map(|x| &**x),
```

Comment:
Try adding `let formatted_id = formatted_id.as_ref().map(|x| &**x);` before the if and then using `formatted_id` here to prevent duplication of this.

Reply:
Ty!  b1b91ce3728ea1eea6e0b490803a037cfeec198a

Reply:
Or 
```suggestion
                                formatted_id.as_deref(),
```

Reply:
Ty!  Let's see what the response is in https://github.com/rust-lang/rust/pull/69155#discussion_r380837981.  If we are going to simply close this I won't push any further changes.  I can make this change if the PR is helpful for someone else to base the additional required work on.

Reply:
Resolving this as it looks like we are heading in a different direction with the approach here.  Thanks for your input here @lzutao!